sw/device/silicon_creator/mask_rom/mask_rom_start.S

```diff
@@ -131,6 +135,30 @@ _mask_rom_start_boot:
   li   t0, 0xFFFF0888
   csrc mie, t0
 
+  // Check if AST initialization should be skipped.
+  li   t0, (TOP_EARLGREY_OTP_CTRL_CORE_BASE_ADDR + \
+            OTP_CTRL_SW_CFG_WINDOW_REG_OFFSET + \
+            OTP_CTRL_PARAM_CREATOR_SW_CFG_AST_INIT_EN_OFFSET)
+  li   t1, MULTIBIT_ASM_BOOL4_TRUE
+  bne  t0, t1, .L_ast_init_skip
```

Comment:
I guess that's the part that do the AST configuration skip based on the INIT_EN tag. As far as I understand, it looks fine.
In DEV/PROD* LC states, ROM or ROM_EXT SW should check that ast_init_done was indeed set. At those LC states, INIT_EN is expected to be true. I think that for the engineering samples, it is better to have this checking code in the ROM_EXT.

Reply:
Created #9685.